### PR TITLE
Sequenced pool synonym for legacy pool

### PR DESCRIPTION
**Pull request by matthew1001**
## PR description
Introduces the new `sequenced` synonym for the `legacy` transaction pool, as discussed in https://github.com/hyperledger/besu/issues/6211.

The PR adds enough for users to start switching to the `--tx-pool=sequenced` option ahead of removal of the `legacy` pool in the future.

I've added one or two additional unit tests to check that `legacy` options still work with the `sequenced` pool, but haven't done anything more comprehensive than that because the future PR to ~~deprecate~~ remove the `legacy` pool will refactor all of the existing tests to use `sequenced` anyway.

## Fixed Issue(s)
Fixes https://github.com/hyperledger/besu/issues/6211

### CODE REVIEW

**Comment by matthew1001** [comment]
Re. docs, I think the https://besu.hyperledger.org/public-networks/concepts/transactions/pool topic probably ought to have a new sub-heading for the sequenced pool. We didn't have one before for the legacy pool but with recent changes to the legacy pool we should try to describe the use-cases the sequenced pool is aimed at and how it differs from the layered pool.

I'd be happy to provide some words to help with that docs topic.

File: besu/src/main/java/org/hyperledger/besu/cli/options/TransactionPoolOptions.java [hunk 1]
```diff
@@ -295,14 +296,14 @@ public void validate(
...
+        "Could not use legacy or sequenced transaction pool options with layered implementation",
         !txPoolImplementation.equals(LAYERED),
-        CommandLineUtils.getCLIOptionNames(Legacy.class));
+        CommandLineUtils.getCLIOptionNames(Sequenced.class));
 
     CommandLineUtils.failIfOptionDoesntMeetRequirement(
         commandLine,
-        "Could not use layered transaction pool options with legacy implementation",
-        !txPoolImplementation.equals(LEGACY),
+        "Could not use layered transaction pool options with legacy or sequenced implementation",
```
Comment (macfarla) [inline besu/src/main/java/org/hyperledger/besu/cli/options/TransactionPoolOptions.java:305]:
```suggestion
        "Could not use layered transaction pool options with legacy (sequenced) implementation",
```

**Review by macfarla — APPROVED** [review]
LGTM . non-blocking suggestion re warning message text - just to make it clearer that they are not separate implementations

Also prob need a changelog entry

File: besu/src/main/java/org/hyperledger/besu/cli/options/TransactionPoolOptions.java (hunk 1 above)
Comment (matthew1001) [reply besu/src/main/java/org/hyperledger/besu/cli/options/TransactionPoolOptions.java:305]: Yeah it's difficult balancing the fact that we don't want to make the sequenced pool sound like it's the "legacy" option, even though in reality they are the same implementation. At some point the legacy option will be removed entirely so we'll update these messages to just talk about sequenced in the future. Personally I'd prefer to leave it as "legacy or sequenced" for now, but happy to discuss further and we can always raise another PR to re-word if we need to?

**Comment by matthew1001** [comment]
> LGTM . non-blocking suggestion re warning message text - just to make it clearer that they are not separate implementations
> 
> Also prob need a changelog entry

Thanks for the comments @macfarla. I've added a reply about the warning message wording. Re. change log entry, I've added a one-liner to it already. Did you want anything more substantial, like Fabio did for the layered pool in `23.10.0`?

**Pull request merged**